### PR TITLE
fix: Image Template Make `customizations` optional - `avm/res/virtual-machine-images/image-template`

### PR DESCRIPTION
**Pull request by AlexanderSehr**
## Description

- The `customizations` property is optional in the API and should also be in the module in case somebody just wants to backup a marketplace image
- Fix for script upload (prep for #3005)

## Pipeline Reference

| Pipeline |
| -------- |
|   [avm.res.virtual-machine-images.image-template](https://github.com/Azure/bicep-registry-modules/actions/workflows/avm.res.virtual-machine-images.image-template.yml)       |

## Type of Change

- [ ] Update to CI Environment or utilities (Non-module affecting changes)
- [ ] Azure Verified Module updates:
  - [x] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

### CODE REVIEW

**Review by ahmadabdalla — DISMISSED** [review]

File: avm/res/virtual-machine-images/image-template/tests/e2e/max/dependencies.bicep [hunk 1]
```diff
@@ -271,11 +271,11 @@ resource assetsStorageAccount_upload 'Microsoft.Resources/deploymentScripts@2023
     scriptContent: loadTextContent('../../../../../../utilities/e2e-template-assets/scripts/Set-StorageContainerContentByEnvVar.ps1')
     environmentVariables: [
       {
-        name: 'script_Install__LinuxPowerShell_sh' // May only be alphanumeric characters & underscores. The upload will replace '_' with '.' and '__' with '-'.
+        name: '__SCRIPT__Install__LinuxPowerShell_sh' // May only be alphanumeric characters & underscores. The upload will replace '_' with '.' and '__' with '-'.
```
Comment (eriqua) [inline avm/res/virtual-machine-images/image-template/tests/e2e/max/dependencies.bicep:274]: Is the updated script used by other modules as well (in other words: should this update happen in other module tests as well)?

Comment (AlexanderSehr) [reply avm/res/virtual-machine-images/image-template/tests/e2e/max/dependencies.bicep:274]: Hey @eriqua, only by this module - and the to-be AIB pattern module.

File: avm/res/virtual-machine-images/image-template/main.bicep [hunk 2]
```diff
@@ -26,10 +26,10 @@ param subnetResourceId string?
 param imageSource object
 
 @description('Required. Customization steps to be run when building the VM image.')
```
[the remark this answers is not in the document]
Comment (AlexanderSehr) [reply avm/res/virtual-machine-images/image-template/main.bicep]: Addressed, and files regenerated. Good find 💪

File: avm/res/virtual-machine-images/image-template/main.bicep [hunk 3]
```diff
@@ -25,7 +25,7 @@ param subnetResourceId string?
 @description('Required. Image source definition in object format.')
 param imageSource object
 
-@description('Required. Customization steps to be run when building the VM image.')
+ @description('Optional. Customization steps to be run when building the VM image.')
```
Comment (eriqua) [inline avm/res/virtual-machine-images/image-template/main.bicep:28]:
```suggestion
@description('Optional. Customization steps to be run when building the VM image.')
```

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**